PSGSuite/Public/Chrome/Get-GSChromeDeviceTelemetry.ps1

```diff
@@ -0,0 +1,137 @@
+    * "storageInfo"
+    * "storageStatusReport"
+    The default value is to return all fields.
+    .PARAMETER Filter
+    Query string for searching Chrome device fields. Supported filter fields are:
+    * "orgUnitId"
+    * "serialNumber"
+    For more information on constructing filter queries, see: https://developers.google.com/admin-sdk/directory/v1/guides/search-users
+    PowerShell filter syntax here is supported as "best effort". Please use Google's filter operators and syntax to ensure best results
+    .PARAMETER PageSize
+    Page size of the result set
+    .PARAMETER Limit
+    The maximum amount of results you want returned. Exclude or set to 0 to return all results
+    .EXAMPLE
+    Get-GSChromeDeviceTelemetry
+    Gets the list of all Chrome devices including all telemetry fields
+    .EXAMPLE
+    Get-GSChromeDeviceTelemetry -Filter "orgUnitId -eq ABCDEFG"
+    Gets the list of all Chrome devices found in the ABCDEFG OU and includes all telemetry fields
+    .EXAMPLE
+    Get-GSChromeDeviceTelemetry -FieldMask deviceId, batteryInfo, cpuInfo
+    Gets the list of all Chrome devices including only the deviceId, batteryInfo and cpuInfo telemetry fields.
+    .LINK
+    https://developers.google.com/chrome/management/reference/rest/v1/customers.telemetry.devices
+    #>
+    [OutputType('Google.Apis.ChromeManagement.v1.Data.GoogleChromeManagementV1TelemetryDevice')]
+    Param
+    (
+        [parameter(Mandatory = $false)]
+        [Alias("Query")]
+        [string]
+        $Filter,
+        [parameter(Mandatory = $false)]
+        [ALias("Fields")]
+        [ValidateSet("name", "customer", "orgUnitId", "deviceId", "serialNumber", "cpuInfo", "cpuStatusReport", "memoryInfo", "memoryStatusReport", "networkStatusReport", "osUpdateStatus", "graphicsInfo", "graphicsStatusReport", "batteryInfo", "batteryStatusReport", "storageInfo", "storageStatusReport")]
+        [String[]]
+        $FieldMask = @("name", "customer", "orgUnitId", "deviceId", "serialNumber", "cpuInfo", "cpuStatusReport", "memoryInfo", "memoryStatusReport", "networkStatusReport", "osUpdateStatus", "graphicsInfo", "graphicsStatusReport", "batteryInfo", "batteryStatusReport", "storageInfo", "storageStatusReport"),
+        [parameter(Mandatory = $false)]
+        [ValidateRange(1,200)]
+        [Alias("MaxResults")]
+        [Int]
+        $PageSize = 200,
+        [parameter(Mandatory = $false)]
+        [Alias('First')]
+        [Int]
+        $Limit = 0
+    )
+    Begin {
+        $serviceParams = @{
+            Scope       = 'https://www.googleapis.com/auth/chrome.management.telemetry.readonly'
+            ServiceType = 'Google.Apis.ChromeManagement.v1.ChromeManagementService'
+        }
+        $service = New-GoogleService @serviceParams
+    }
+    Process {
+                $verbString = "Getting all Chrome device telemetry records"
+                try {
+                    $request = $service.Customers.Telemetry.Devices.List('customers/my_customer')
```

Comment:
I noticed this as I was looking through again.  According to the API/.Net documentation, a value of `my_customer` is required to get devices for the current customer, or passing in a CustomerID to get devices for that Customer.

I know the documentation is not always 100% correct, but I'm wondering if this works with just `my_customer` vs `customers/my_customer`?  Would it make sense to have a CustomerID parameter that defaults to `my_customer` (or whatever default parameter will return the current customer)?

Reply:
Good question.

From my testing the value needs to be prefixed with `customers/`. When using only `my_customer` the following output is produced.
```
Get-GSChromeDeviceTelemetry : Exception calling "Execute" with "0" argument(s): "Parameter validation failed for "parent" : The value did not match the regular expression ^customers/[^/]+$"
```

I have tested using my actual CustomerID `customers/<my_Customer_ID>` and that succeeded. So I think it would be fair to say that yes we could add a parameter for the CustomerID if a practical usage of the parameter can be identified.

I am unfamiliar with the use of CustomerIDs and am unsure how they would be used or how the value changes in practice?

Am I correct in my assumption that the CustomerID uniquely identifies your domain? So if for example you had a child domain attached to the same account then it would have a different CustomerID from the primary domain?

If the above is correct then it would make sense to give the option to change the CustomerID to allow accessing the data from the child domain.

Let me know if I am understanding things correctly here and I will be happy to add the parameter.

Reply:
To be honest I'm not sure how it works exactly, as I'm not a Google admin in my org.

From what you've said it sounds like the documentation is wrong and it does require the customer id to be prepended with `customers/`.  So maybe a parameter of `CustomerID` that defaults to `my_customer` and when the call is actually made, it uses `customers/$CustomerID`.  I think that would be functionally identical to what you've got now: default to the user making the call, but then also add the option to specify a CustomerID, since it's possible to do so with the API.

Reply:
I have now added the $CustomerID parameter as discussed above.